src/pages/reports/collectionIdSearch.js

```diff
@@ -0,0 +1,76 @@
+import { masterSpecimenIDRequirement } from '../../tubeValidation.js';
+import { finalizeTemplate } from '../finalize.js';
+import fieldToConceptIdMapping from "../../fieldToConceptIdMapping.js";
+
+export const collecionIdSearchScreen = async (auth, route) => {
+    const user = auth.currentUser;
+    if (!user) return;
+    const username = user.displayName ? user.displayName : user.email;
+    appState.setState({ 'username': username });
+    collecionIdSearchScreenTemplate(username);
+};
+
+export const collecionIdSearchScreenTemplate = async (username, auth, route) => {
+    let template = "";
+    template += reportsNavbar();
+    template += ` 
+              <div id="root root-margin" style="padding-top: 25px;">
+                <h3 style="text-align: center; margin: 1rem 0;">Collection ID Search</h3>
+                <div class="row">
+                    <div class="col-lg">
+                        <div class="row form-row">
+                            <form id="specimenLookupForm" method="POST">
+                                <div class="form-group">
+                                    <label class="search-label">Collection ID</label>
+                                    <input class="form-control" autocomplete="off" required type="text" id="masterSpecimenId" placeholder="Enter/Scan Collection ID"/>
+                                </div>
+                                <div class="form-group">
+                                    <button type="submit" class="btn btn-outline-primary">Search</button>
+                                </div>
+                            </form>
+                        </div>
+                    </div>
+                </div>
+              </div>
+  `;
+
+    document.getElementById("contentBody").innerHTML = template;
+    document.getElementById("navbarNavAltMarkup").innerHTML = nonUserNavBar(username);
+    activeReportsNavbar();
+    searchSpecimenEvent();
+};
+
+const searchSpecimenEvent = () => {
+    const form = document.getElementById('specimenLookupForm');
+    if (!form) return;
+    form.addEventListener('submit', async e => {
+        e.preventDefault();
+        removeAllErrors();
+        let masterSpecimenId = document.getElementById('masterSpecimenId').value.toUpperCase();
+
+        if(masterSpecimenId.length > masterSpecimenIDRequirement.length) masterSpecimenId = masterSpecimenId.substring(0, masterSpecimenIDRequirement.length);
+
+        if (!masterSpecimenIDRequirement.regExp.test(masterSpecimenId) || masterSpecimenId.length !== masterSpecimenIDRequirement.length) {
+            errorMessage('masterSpecimenId', `Collection ID must be ${masterSpecimenIDRequirement.length} characters long and in CXA123456 format.`, true);
+            return;
+        }
+        showAnimation();
+        const biospecimen = await searchSpecimen(masterSpecimenId);
```

Comment:
Correct me If I am wrong, but should there be another endpoint that needs to be created or modify an existing one on connectFaas? The `searchSpecimen` is currently site dependent. I can search and get NIH collection IDs but not other sites. Do we need to access all biospecimen documents and filter them based on the collection ID?

biospecimen repo: [searchSpecimen function](https://github.com/episphere/biospecimen/blob/dev/src/shared.js#L804-L813)

connectFass repo: 
[endpoint ](https://github.com/episphere/connectFaas/blob/dev/utils/biospecimen.js#L210-L229)
[searchSpecimen function](https://github.com/episphere/connectFaas/blob/dev/utils/firestore.js#L970-L980)